go-controller/pkg/util/pod_annotation.go

```diff
@@ -210,3 +211,33 @@ func UnmarshalPodAnnotation(annotations map[string]string) (*PodAnnotation, erro
 
 	return podAnnotation, nil
 }
+
+// GetAllPodIPs returns the pod's IP addresses, first from the OVN annotation
+// and then falling back to the Pod Status IPs. This function is intended to
+// also return IPs for HostNetwork and other non-OVN-IPAM-ed pods.
+func GetAllPodIPs(pod *v1.Pod) ([]net.IP, error) {
+	annotation, err := UnmarshalPodAnnotation(pod.Annotations)
+	if annotation != nil {
+		// Use the OVN annotation if valid
+		ips := make([]net.IP, 0, len(annotation.IPs))
+		for _, ip := range annotation.IPs {
+			ips = append(ips, ip.IP)
+		}
+		return ips, nil
+	}
+
+	if len(pod.Status.PodIPs) == 0 {
+		return nil, err
+	}
+
+	// Otherwise if the annotation is not valid try to use Kube API pod IPs
+	ips := make([]net.IP, 0, len(pod.Status.PodIPs))
+	for _, podIP := range pod.Status.PodIPs {
+		ip := net.ParseIP(podIP.IP)
+		if ip == nil {
+			klog.Warningf("failed to parse pod IP %q", podIP)
+		}
+		ips = append(ips, ip)
```

Comment:
clarification: do we want to have add `nil` values to slice of `ips`? As long as the caller of this function is ensuring that the slice contain nil value and is checking for it, it could be OK

Reply:
sorry Vish I just missed this as I merged it. If there is something to improve here lets open another PR.

Reply:
@trozet no worries, it was a clarification question as I am writing unit tests for this file currently. Yes, it can be handled in another PR if there are issues